lighthouse-core/gather/connections/connection.js

```diff
@@ -19,13 +19,31 @@
 const EventEmitter = require('events').EventEmitter;
 const log = require('../../lib/log.js');
 
+/* eslint-disable no-unused-vars */
+
+/**
+ * @interface
+ */
+class ConnectionDelegate {
+  /**
+   * Evaluate an expression in the context of the current page.
+   * Returns a promise that resolves on the expression's value.
+   * @param {string} expression
+   * @return {!Promise<*>}
+   */
+  evaluateAsync(expression) { }
```

Comment:
> this function already exists in driver, rather than implement this in each connection type thoughts on just using the userAgent method everywhere for now?

yeah, I like this. Not sure why I skimmed over @paulirish suggesting that in #1927 :)

`driver.evaluateAsync('/Chrome\\/([0-9.]+)/.exec(navigator.userAgent)[1]')` would work (have to double escape the `/`, arguably a bug in our current `evaluateAsync`), or just `driver.evaluateAsync('navigator.userAgent')` if you want to do the regexing on the node side